### PR TITLE
feat: buy flow wording adjusted

### PR DESCRIPTION
**Pull request by anastasiyaig**
### What does the PR do

Fix https://github.com/status-im/status-desktop/issues/16469

Address feedback from legal https://discord.com/channels/1210237582470807632/1217171454940938341/1291353246874734622

### Affected areas

Buy crypto modal

### Screenshots

Screenshot 2024-11-25 at 10 05 46 PM

Screenshot 2024-11-25 at 9 24 30 PM

Screenshot 2024-11-25 at 9 05 16 PM

Screenshot 2024-11-25 at 9 09 15 PM

### CODE REVIEW

**Comment by anastasiyaig** [comment]
I am a bit confused , we dont have **Buy assets** button, but we have **Buy crypto** , also the styling of that component is different

I changed Buy to Add which i suppose is the main idea of this renaming, shall crypto be changed to assets?

Figma wants this:
image

We have now this:
Screenshot 2024-11-25 at 6 59 04 PM

**Review by alaibe — APPROVED** [review]

File: ui/app/AppLayouts/Wallet/WalletLayout.qml [hunk 1]
```diff
@@ -367,7 +367,6 @@ Item {
             onLaunchBuyCryptoModal: {
                 d.buyFormData.selectedWalletAddress = d.getSelectedOrFirstNonWatchedAddress()
                 d.buyFormData.selectedNetworkChainId = StatusQUtils.ModelUtils.getByKey(RootStore.filteredFlatModel, "layer", 1, "chainId")
-                d.buyFormData.selectedTokenKey = Constants.ethToken
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/WalletLayout.qml:370]: The idea here is to be able distinguish between explicitly buying ETH or just any token; we can make the fallback later in the BuyCryptoModal...

File: ui/app/AppLayouts/Wallet/popups/buy/BuyCryptoModal.qml [hunk 2]
```diff
@@ -88,7 +88,7 @@ StatusStackModal {
         readonly property ModelEntry selectedTokenEntry: ModelEntry {
             sourceModel: root.plainTokensBySymbolModel
             key: "key"
-            value: root.buyCryptoInputParamsForm.selectedTokenKey
+            value: root.buyCryptoInputParamsForm.selectedTokenKey || Constants.ethToken
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/buy/BuyCryptoModal.qml:91]: ... here, so we can see whether we explicitly requested to buy some specific token, or just any token

**Review by Khushboo-dev-cpp — APPROVED** [review]
Tested and  LGTM!
What's done is
1. Whenever the token is decipherable we show the Token Name in the Buy Modal
2. When not possible we show "`Ways to Buy assets`"
3. In Swap Modal when there is no gas we show `"Not enough ETH to pay gas fees" `and `"Add Eth"`
4.  When there is not enough tokens for swap we show `"Insufficient funds for swap"` and `"Add assets"`

**Comment by anastasiyaig** [comment]
user story created here https://www.notion.so/Buy-flow-wording-adjusted-1428f96fb65c80b8bc9dcccd8eb092c8?pvs=4

**Pull request merged**